File: src/c#/Creational/FactoryMethod/FactoryMethod1/ManagerFactory.cs
using System;
using Util;

namespace FactoryMethod1
{
    public class ManagerFactory : IFactory
    {
        public Employee Create() { 
            string department = EmployeeRandomizer.CreateDepartment();
            Manager manager = new Manager(EmployeeRandomizer.CreateId(), EmployeeRandomizer.CreateName(),
                EmployeeRandomizer.CreateYear(), department, department);
            return manager;
        }
    }
}

File: src/typescript/com/gokselkucuksahin/dp/creational-patterns/factory/employee1/manager-factory.ts
import {Factory} from "./factory";
import {Manager} from "./manager";
import {EmployeeRandomizer} from "../employee-randomizer";

export class ManagerFactory implements Factory {
  create(): Manager {
    const department = EmployeeRandomizer.createDepartment();
    return new Manager(EmployeeRandomizer.createId(), EmployeeRandomizer.createName(),
        EmployeeRandomizer.createYear(), department, department);
  }
}